Tests/TensorFlowTests/OperatorTests/BasicTests.swift

```diff
@@ -443,19 +443,60 @@ final class BasicOperatorTests: XCTestCase {
         XCTAssertEqual(result.shape, [1, 3, 1, 2, 1])
     }
 
-    func testUnbroadcast1() {
+    func testUnbroadcastRank4ToRank2() {
         let x = Tensor<Float>(repeating: 1, shape: [2, 3, 4, 5])
         let y = Tensor<Float>(repeating: 1, shape: [4, 5])
         let z = x.unbroadcasted(like: y)
         XCTAssertEqual(z.array, ShapedArray<Float>(repeating: 6, shape: [4, 5]))
     }
 
-    func testUnbroadcast2() {
+    func testUnbroadcastRank4ToRank3() {
         let x = Tensor<Float>(repeating: 1, shape: [2, 3, 4, 5])
         let y = Tensor<Float>(repeating: 1, shape: [3, 1, 5])
         let z = x.unbroadcasted(like: y)
         XCTAssertEqual(z.array, ShapedArray<Float>(repeating: 8, shape: [3, 1, 5]))
     }
+    
+    func testUnbroadcast3x3To1x3() {
+        func foo(tensor: Tensor<Float>, shape: Tensor<Int32>) -> Tensor<Float> {
+            tensor.unbroadcasted(toShape: shape)
+        }
+        var inputTensor: Tensor<Float>
+        var expected: Tensor<Float>
+        var pb: (Tensor<Float>) -> Tensor<Float>
+        
+        // [3,3] -> [1,3]
+        let atTensor: Tensor<Float> = Tensor([
+            [1, 2, 3],
+            [1, 2, 3],
+            [1, 2, 3]]
+        )
+        pb = pullback(at: atTensor) { x in
+            foo(tensor: x, shape: Tensor([1, 3]))
+        }
+        
+        // Same shape as parameter of pullback
+        inputTensor = Tensor([[1, 2, 3]])
```

Comment:
```suggestion
        inputTensor = [[1, 2, 3]]
```
Calling `Tensor` initializer is not necessary when the contextual type is `Tensor`. Just use literal conversion. Same everywhere.